### ISSUE TITLE
Feature: Tidy up the generated production build

### ISSUE THREAD

**Issue by Perezmarc** [angular/angular-cli#3717]
Hi,
I get too many files that I believe are unnecessary (but maybe they're not). Would be nice to have all the production deployment options such as: How to deploy gzipped version, how to deploy not gzipped version, with sourcemaps, without sourcemaps...

This are all the files I get when I execute that command. Would appreciate a way to tidy up the production build, maybe include folders to separate files, such as 'js' & 'css'. I also would like to know if I can upload to my cloudfront the .gz without uploading the .js

![captura de pantalla 2016-12-21 a les 10 34 17](https://cloud.githubusercontent.com/assets/9431228/21451181/5ecb28c2-c8fd-11e6-9b97-7eca7bf6b126.png)

my angular-cli.json

{
"project": {
"version": "1.0.0-beta.22-1",
"name": "webapp-21b"
},
"apps": [
{
"root": "src",
"outDir": "dist",
"assets": [
"assets",
"download",
".well-known",
"robots.txt",
"sitemap.xml"
],
"index": "index.html",
"main": "main.ts",
"test": "test.ts",
"tsconfig": "tsconfig.json",
"prefix": "app",
"mobile": false,
"styles": [
"../node_modules/bootstrap/dist/css/bootstrap.css",
"./css/main.scss",
"./css/buttons.scss",
"./css/forms.scss",
"./css/navbar.scss",
"./css/nouislider.scss",
"./css/mixins.scss",
"./css/spinner.scss",
"./css/utils.scss",
"./css/vars.scss"
],
"scripts": [
"./assets/js/noUiSlider.9.0.0/nouislider.min.js",
"./assets/js/wNumb.js"
],
"environments": {
"source": "environments/environment.ts",
"dev": "environments/environment.ts",
"prod": "environments/environment.prod.ts"
}
}
],
"addons": [
],
"packages": [
],
"e2e": {
"protractor": {
"config": "./protractor.conf.js"
}
},
"test": {
"karma": {
"config": "./karma.conf.js"
}
},
"defaults": {
"styleExt": "scss",
"prefixInterfaces": false,
"inline": {
"style": false,
"template": false
},
"spec": {
"class": false,
"component": true,
"directive": true,
"module": false,
"pipe": true,
"service": true
}
}
}

**Comment by Meligy** [comment]
The only files you may not want there are the `.gz` and `.map` files.

You can provide `--sourcemap false ` (`-sm false` for short) to avoid generating `.map` files.

I don't think there's currently a feature to turn off `.gz` files.

**Comment by Meligy** [comment]
Actually I was wrong about `.gz` files.

There's a flag to turn them off.

`--suppress-sizes true` will do it. It does not have a short form unfortunately.

And there seems to be a bug with it.

![image](https://cloud.githubusercontent.com/assets/171637/21451479/7c6bffb8-c953-11e6-9bd5-6dba81bf431a.png)

It does not affect the `vendor` bundle (it does affect the main bundle).

A workaround is to pass `--vendor-chunk false` (again, no short form).  
Which will take slightly longer, but might also give slightly more optimized bundle (not 100% sure).

Output (assuming no `assets` or fonts etc referenced -- based on beta 24):

![image](https://cloud.githubusercontent.com/assets/171637/21451542/f75c2022-c953-11e6-8342-0a7fa4ca7d33.png)


The command used for this output is:

```
ng build -prod -sm false --suppress-sizes true --vendor-chunk false
```

**Comment by Perezmarc** [comment]
Thanks!

**Comment by filipesilva** [comment]
Closing as @Meligy already gave a great answer, cheers!

**Comment by Meligy** [comment]
@filipesilva there's still a bug with `--suppress-sizes` not working with `vendor` file.

That's the only open part from this issue.

**Comment by filipesilva** [comment]
I actually have no idea what `suppress-sizes` is supposed to do. As far as I can tell, it's an option that's listed but never did anything. Also https://github.com/angular/angular-cli/issues/1841

**Comment by Meligy** [comment]
@filipesilva from what I can see, it eliminates generating `.gz` files in the production build. It does work with main and chunk bundles, but not vendor bundles for some reason.

**Comment by Meligy** [comment]
Actually, I take that back. I tried all the commands again, and agree with @filipesilva , `--supress-sizes` seems to do nothing (explains why I couldn't track it when I was looking for how to supress `.gz` from `vendor`).

The behaviour that confused me was switching on and off vendor chunks at the same time.

With beta 24:

- If you have vendor on, which is the default, you only get a `.gz` file for this vendor chunk. You don't get it for main.
- If you have vendor off, you get a `.gz` file for the main bundle. You don't get it for the other chunks (if you use lazy loading and have other chunks).

This is interesting. It's pretty much what was asked to have in the issue anyway, except now I wonder "Where are my `.gz`s?" as I think we used to have a few more `.gz` files generated.

So, either delete the remaining `.gz` file, or bring back all of them, ideally with a flag to disable them, something like what I thought `--suppress-sizes` did but it doesn't.

One final word about `--suppress-sizes`. I think in the very very old days of the CLI, it used to output sizes to the terminal, it's probably a safe to delete now.
